### PR TITLE
lib: nrf_modem: Revert modem trace change from #6471

### PR DESCRIPTION
**Pull request by gregersrygg**
It was assumed that the modem trace was always enabled at boot, but the
modem trace configuration is persisted in the modem periodically and when
using AT+CFUN=0. This caused a regression when the modem did not already
have the modem trace enabled in the persisted configuration. This commit
reverts the changes in lib/lte_link_control/lte_lc.c from #6471 and
corrects the documentation accordingly. Then modem traces should always
be enabled when CONFIG_NRF_MODEM_LIB_TRACE_ENABLED is turned on.

Fixes NCSDK-13500

Signed-off-by: Gregers Gram Rygg <gregers.gram.rygg@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/libraries/modem/nrf_modem_lib.rst
Comment (balaji-nordic) [inline doc/nrf/libraries/modem/nrf_modem_lib.rst:34]:
```suggestion
When :kconfig:`CONFIG_NRF_MODEM_LIB_TRACE_ENABLED` Kconfig option is enabled, the modem traces are enabled in the modem and are forwarded to the `Modem trace module`_.
```

**Review by balaji-nordic — CHANGES_REQUESTED** [review]

**Review by tokangas — APPROVED** [review]

**Review by balaji-nordic — APPROVED** [review]

**Comment by lemrey** [comment]
Please wait for @eivindj-nordic review before merging.

**Review by eivindj-nordic — APPROVED** [review]

**Review by umapraseeda — APPROVED** [review]

**Pull request merged**
